pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -340,23 +354,17 @@ func arePublicKeysEqual(a, b []*plugintypes.JWTKey) bool {
 	return true
 }
 
-func parseBundle(b *types.Bundle) *plugintypes.Bundle {
+func parseBundle(b *types.Bundle) (*plugintypes.Bundle, error) {
 	typesBundle := proto.Clone(b).(*types.Bundle)
 
-	var jwtAuthorities []*plugintypes.JWTKey
-	for _, jwtauthority := range typesBundle.JwtAuthorities {
-		jwtAuthorities = append(jwtAuthorities, &plugintypes.JWTKey{
-			PublicKey: jwtauthority.PublicKey,
-			KeyId:     jwtauthority.KeyId,
-			ExpiresAt: jwtauthority.ExpiresAt,
-		})
+	jwtAuthorities, err := jwtkey.ToPluginFromAPIProtos(b.JwtAuthorities)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "invalid JWT authority: %v", err)
 	}
 
-	var x509Authorities []*plugintypes.X509Certificate
-	for _, x509Authority := range typesBundle.X509Authorities {
-		x509Authorities = append(x509Authorities, &plugintypes.X509Certificate{
-			Asn1: x509Authority.Asn1,
-		})
+	x509Authorities, err := x509certificate.ToPluginFromAPIProtos(typesBundle.X509Authorities)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "invalid X.509 authority: %v", err)
 	}
 
 	return &plugintypes.Bundle{
```

Comment:
We should probably have a coretypes for bundles that does these conversions... there are several places that do this in the code. We can add it in a later commit though.